src/java.base/share/classes/java/lang/Math.java

```diff
@@ -2191,6 +2191,123 @@ public static double min(double a, double b) {
         return (a <= b) ? a : b;
     }
 
+    /**
+     * Clamps the value to fit between min and max. If the value is less
+     * than {@code min}, then {@code min} is returned. If the value is greater
+     * than {@code max}, then {@code max} is returned. Otherwise, the original
+     * value is returned.
+     * <p>
+     * While the original value of type long may not fit into the int type,
+     * the bounds have the int type, so the result always fits the int type.
+     * This allows to use method to safely cast long value to int with
+     * saturation.
+     *
+     * @param value value to clamp
+     * @param min minimal allowed value
+     * @param max maximal allowed value
+     * @return a clamped value that fits into {@code min..max} interval
+     * @throws IllegalArgumentException if {@code min > max}
+     *
+     * @since 21
+     */
+    public static int clamp(long value, int min, int max) {
```

Comment:
There should probably also be a pure `int` overload:
```java
public static int clamp(int value, int min, int max)
```

Reply:
@ExE-Boss which problem such an overload would solve? It looks like, `int clamp(long, int, int)` can be used everywhere where proposed `int clamp(int, int, int)` could be useful.

Reply:
A pure `int`‑only overload doesn’t have to go through the `int` → `long` → `int` conversion.

Reply:
Is this a real problem?

Reply:
This should not be a problem with C2, and with the lower compilation levels this insignificant impact seems to not justify adding an overload